modules/channel-web/src/views/web/index.jsx

```diff
@@ -118,9 +120,35 @@ export default class Web extends React.Component {
     this.isUnmounted = true
   }
 
+  updateAxiosConfig() {
+    const { botId, externalAuthToken } = this.state.config
+
+    this.axiosConfig = botId
+      ? { baseURL: `${window.location.origin}/api/v1/bots/${botId}` }
+      : { baseURL: `${window.BOT_API_PATH}` }
+
+    if (externalAuthToken) {
+      this.axiosConfig = {
+        ...this.axiosConfig,
+        headers: {
+          ExternalAuth: `Bearer ${externalAuthToken}`
+        }
+      }
+    }
+  }
+
+  changeUserId = newId => {
+    this.props.bp.events.updateVisitorId(newId)
+    this.setState({ currentConversationId: null })
+    this.setUserId().then(this.fetchData)
+  }
+
   handleIframeApi = ({ data: { action, payload } }) => {
     if (action === 'configure') {
-      this.setState({ config: Object.assign({}, defaultOptions, payload) })
+      if (payload.userId) {
+        this.changeUserId(payload.userId)
+      }
+      this.setState({ config: Object.assign({}, defaultOptions, payload) }, this.updateAxiosConfig)
```

Comment:
you're using object spread in updateAxios config . why not use it here for a matter of consistency 
```js
{ config : { ...defaultOptions, ...payload }}
```